src/js/tracing/reactnativenavigation.ts

```diff
@@ -0,0 +1,206 @@
+import { Transaction as TransactionType } from "@sentry/types";
+import { logger } from "@sentry/utils";
+import { EmitterSubscription } from "react-native";
+
+import { BeforeNavigate } from "./reactnativetracing";
+import {
+  RoutingInstrumentation,
+  TransactionCreator,
+} from "./routingInstrumentation";
+import { getBlankTransactionContext } from "./utils";
+
+interface ReactNativeNavigationOptions {
+  routeChangeTimeoutMs: number;
+}
+
+const defaultOptions: ReactNativeNavigationOptions = {
+  routeChangeTimeoutMs: 1000,
+};
+
+interface ComponentEvent {
+  componentId: string;
+}
+
+type ComponentType =
+  | "Component"
+  | "TopBarTitle"
+  | "TopBarBackground"
+  | "TopBarButton";
+
+export interface ComponentWillAppearEvent extends ComponentEvent {
+  componentName: string;
+  passProps?: Record<string | number | symbol, unknown>;
+  componentType: ComponentType;
+}
+
+export interface EventSubscription {
+  remove(): void;
+}
+
+export interface EventsRegistry {
```

Comment:
Can we hook into https://wix.github.io/react-native-navigation/api/events#onapplaunched somehow?